hangman.py

```diff
@@ -0,0 +1,217 @@
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    for letter in secret_word:
+        if letter in letters_guessed:
+            continue
+        else:
+            return False
+    return True
+
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    t_word = []
+    for i in secret_word:
+        if i in letters_guessed:
+            t_word.append(i)
+        else:
+            t_word.append("_ ")
+    return ''.join(t_word)
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    all_letters = string.ascii_lowercase
+    available_letters = []
+    for i in range(len(all_letters)):
+        if all_letters[i] not in letters_guessed:
+            available_letters.append(all_letters[i])
+    return ''.join(available_letters)
+
+
+def hangman(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+
+    guesses_remaining = 6
+    warnings_remaining = 3
+    letters_guessed = []
+    print(f'''Welcome to the game Hangman!
+I am thinking of a word that is {len(secret_word)} letters long.
+You have {warnings_remaining} warnings left''')
+    while (guesses_remaining > 0) and not (is_word_guessed(secret_word, letters_guessed)):
+        print(f'''-------------
+You have {guesses_remaining} guesses left.
+Available letters: {get_available_letters(letters_guessed)}''')
+
+        letters_guessed.append((input("Please guess a letter:")).lower())
```

Comment:
I believe it would be better to check the input first, and append only a valid input

Reply:
+